tests/test_response_dict_by_alias.py

```diff
@@ -0,0 +1,58 @@
+from fastapi import Depends, FastAPI
+from fastapi.testclient import TestClient
+from pydantic import BaseModel, Field
+
+app = FastAPI()
+
+
+class MockServiceModel(BaseModel):
+    full_name: str = Field(alias="F_NAME")
+
+
+class MockService:
+    def __init__(self):
+        self.initialized = True
+        self.data = {"F_NAME": "Ilyas Q."}
+
+    def get_data(self) -> MockServiceModel:
+        return MockServiceModel(**self.data)
+
+
+class ResponseSchema(BaseModel):
+    full_name: str = Field(alias="fullName")
+
+    class Config:
+        allow_population_by_field_name = True
+
+
+def get_mock_service():
+    return MockService()
+
+
+@app.get("/by-alias")
+def by_alias(service: MockService = Depends(get_mock_service)):
+    data = service.get_data()
+
+    return data
+
+
+@app.get("/by-field-name", response_model=ResponseSchema, response_dict_by_alias=False)
+def by_field_name(service: MockService = Depends(get_mock_service)):
+    data = service.get_data()
+
+    return data
+
+
+client = TestClient(app)
+
+
+def test_response_dict_by_alias():

```

Comment:
I'd recommend adding a test that you know will fail e.g. asserting the status code to be 422 for a validation error. The example you give in your summary would be good to replicate as this would cover any changes to the underlying functionality e.g., if, without setting `response_dict_by_alias` to true, we can set by dict name rather than alias then we should be notified as some changes might have made this bool redundant or some underlying logic has been changed which has some unwanted side effects.